Uploading a document with failed conversion still triggers segmentation

`3:10 PM: 2025-05-04T15:10:24.259Z [planinternational] requestId: 1928 
Error: pdftotext /tmp/1746371423332hm58bhyiutt.pdf - failed with code 1
stderr output:
Syntax Warning: May not be a PDF file (continuing anyway)
Syntax Error (2): Illegal character <21> in hex string
Syntax Error (4): Illegal character <6f> in hex string`

`2025-05-04 15:10:24,284 [ERROR] {"tenant":"planinternational","task":"segmentation","params":{"filename":"1746371423332hm58bhyiutt.pdf","language":"en"},"success":false,"error_message":"Error processing PDF","data_url":null,"file_url":null}
Traceback (most recent call last):
  File "/app/src/drivers/queues_processor/start_queue_processor.py", line 65, in process
    return process_task(task).model_dump_json()
           ^^^^^^^^^^^^^^^^^^
  File "/app/src/drivers/queues_processor/start_queue_processor.py", line 81, in process_task
    extraction_data = extract_segments(task, xml_file_name)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/src/use_cases/extract_segments_use_case.py", line 66, in extract_segments
    raise RuntimeError(f"Error processing PDF document: {results.status_code} - {results.text}")
RuntimeError: Error processing PDF document: 422 - {"detail":"Error see traceback"}`

--- a/app/api/services/pdfsegmentation/PDFSegmentation.ts
+++ b/app/api/services/pdfsegmentation/PDFSegmentation.ts
@@ -87,6 +87,7 @@ class PDFSegmentation {
       {
         type: 'document',
         filename: { $exists: true },
+        status: 'ready',
         _id: { $nin: segmentedFiles },
       },
       'filename',

--- a/app/api/services/pdfsegmentation/specs/PDFSegmentation.spec.ts
+++ b/app/api/services/pdfsegmentation/specs/PDFSegmentation.spec.ts
@@ -1,3 +1,4 @@
+/* eslint-disable max-statements */
 /* eslint-disable camelcase */
 /* eslint-disable max-lines */
 
@@ -68,13 +69,13 @@ describe('PDFSegmentation', () => {
   let fileA: Buffer;
   let fileB: Buffer;
 
-  afterAll(async () => {
-    await DB.disconnect();
-  });
-
   beforeAll(async () => {
     const uri = config.DBHOST;
     await DB.connect(`${uri}PDFSegmentation_spec`);
+  });
+
+  afterAll(async () => {
+    await DB.disconnect();
   });
 
   beforeEach(async () => {
@@ -150,7 +151,7 @@ describe('PDFSegmentation', () => {
     }, 'tenantOne');
   });
 
-  it('should only send pdfs not already segmented or in the process', async () => {
+  it('should only send ready pdfs not already segmented or in the process', async () => {
     await fixturer.clearAllAndLoad(dbOne, fixturesFiveFiles);
     await dbOne.collection('segmentations').insertMany([
       {

--- a/app/api/services/pdfsegmentation/specs/fixtures.ts
+++ b/app/api/services/pdfsegmentation/specs/fixtures.ts
@@ -31,19 +31,40 @@ const fixturesPdfNameB = 'documentB.pdf';
 const fixturesOneFile: DBFixture = {
   entities: [factory.entity('A1', 'templateToSegmentA')],
   settings,
-  files: [factory.fileDeprecated('F1', 'A1', 'document', fixturesPdfNameA)],
+  files: [
+    factory.file('F1', {
+      filename: fixturesPdfNameA,
+      entity: 'A1',
+      type: 'document',
+      status: 'ready',
+    }),
+  ],
 };
 
 const fixturesOtherFile: DBFixture = {
   entities: [factory.entity('A2', 'templateToSegmentB')],
   settings: otherSettings,
-  files: [factory.fileDeprecated('F2', 'A2', 'document', fixturesPdfNameB)],
+  files: [
+    factory.file('F2', {
+      filename: fixturesPdfNameB,
+      entity: 'A2',
+      type: 'document',
+      status: 'ready',
+    }),
+  ],
 };
 
 const fixturesMissingPdf: DBFixture = {
   entities: [factory.entity('A1', 'templateToSegmentA')],
   settings,
-  files: [factory.fileDeprecated('F1', 'A1', 'document', 'missing.pdf')],
+  files: [
+    factory.file('F1', {
+      filename: 'missing.pdf',
+      entity: 'A1',
+      type: 'document',
+      status: 'ready',
+    }),
+  ],
 };
 
 const fixturesFiveFiles: DBFixture = {
@@ -56,11 +77,42 @@ const fixturesFiveFiles: DBFixture = {
     factory.entity('A5', 'templateToSegmentA'),
   ],
   files: [
-    factory.fileDeprecated('F1', 'A1', 'document', fixturesPdfNameA),
-    factory.fileDeprecated('F2', 'A2', 'document', fixturesPdfNameA),
-    factory.fileDeprecated('F3', 'A3', 'document', fixturesPdfNameA),
-    factory.fileDeprecated('F4', 'A4', 'document', fixturesPdfNameA),
-    factory.fileDeprecated('F5', 'A5', 'document', fixturesPdfNameA),
+    factory.file('F1', {
+      filename: fixturesPdfNameA,
+      entity: 'A1',
+      type: 'document',
+      status: 'ready',
+    }),
+    factory.file('F2', {
+      filename: fixturesPdfNameA,
+      entity: 'A2',
+      type: 'document',
+      status: 'ready',
+    }),
+    factory.file('F3', {
+      filename: fixturesPdfNameA,
+      entity: 'A3',
+      type: 'document',
+      status: 'ready',
+    }),
+    factory.file('F4', {
+      filename: fixturesPdfNameA,
+      entity: 'A4',
+      type: 'document',
+      status: 'ready',
+    }),
+    factory.file('F5', {
+      filename: fixturesPdfNameA,
+      entity: 'A5',
+      type: 'document',
+      status: 'ready',
+    }),
+    factory.file('F6', {
+      filename: fixturesPdfNameA,
+      entity: 'A5',
+      type: 'document',
+      status: 'failed',
+    }),
   ],
 };
 
@@ -70,7 +122,12 @@ const fixturesOneHundredFiles: DBFixture = {
     factory.entity(`A${x.toString()}`, 'templateToSegmentA')
   ),
   files: [...Array(100).keys()].map(x =>
-    factory.fileDeprecated(`F${x.toString()}`, `A${x.toString()}`, 'document', fixturesPdfNameA)
+    factory.file(`F${x.toString()}`, {
+      filename: fixturesPdfNameA,
+      entity: `A${x.toString()}`,
+      type: 'document',
+      status: 'ready',
+    })
   ),
 };
 

